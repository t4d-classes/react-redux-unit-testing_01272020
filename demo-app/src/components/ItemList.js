import React from 'react';

export const ItemList = ({ items }) => {
  return <ul>
    {items.map( (item, index) => <li key={index}><div>{item}</div></li>)}
  </ul>;
};
