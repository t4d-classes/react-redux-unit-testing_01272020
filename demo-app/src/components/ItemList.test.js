import React from 'react';
import { render } from 'enzyme';

import { ItemList } from './ItemList';

describe('ItemList Snapshot Tests', () => {

  test('ItemList displays Hello World', () => {

    const colors = ['red', 'green', 'blue', 'brown', 'yellow'];

    const itemListHtml = render(
      <ItemList items={colors} />
    ).html()

    const component = JSON.stringify(itemListHtml);

    expect(component).toMatchSnapshot();

  });

});