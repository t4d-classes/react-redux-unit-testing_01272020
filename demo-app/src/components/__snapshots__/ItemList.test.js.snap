// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemList Snapshot Tests ItemList displays Hello World 1`] = `"\\"<li><div>red</div></li><li><div>green</div></li><li><div>blue</div></li><li><div>brown</div></li><li><div>yellow</div></li>\\""`;
